src/fs/file.rs

```diff
@@ -491,31 +515,32 @@ impl File {
             );
         }
 
+        let (in_begin, in_end) = (buf.begin(), buf.end());
         let mut bytes_written = 0;
         while bytes_written < buf_len {
-            let (res, slice) = self
-                .write_at(buf.slice(bytes_written..), pos + bytes_written as u64)
-                .await;
-            buf = slice.into_inner();
+            let (res, slice) = self.write_at(buf, pos + bytes_written as u64).await;
             match res {
                 Ok(0) => {
                     return (
                         Err(io::Error::new(
                             io::ErrorKind::WriteZero,
                             "failed to write whole buffer",
                         )),
-                        buf,
+                        slice.into_inner().slice(in_begin..in_end),
                     )
                 }
                 Ok(n) => {
                     bytes_written += n;
+                    buf = slice.slice(n..);
                 }
-                Err(ref e) if e.kind() == io::ErrorKind::Interrupted => {}
-                Err(e) => return (Err(e), buf),
+                Err(ref e) if e.kind() == io::ErrorKind::Interrupted => {
```

Comment:
I wonder if `EINTR` should occur with the way this runtime uses io-uring. The `write_all` implementations for socket streams says it does not happen.

Reply:
@FrankReh please comment whether this match arm is needed at all, or is it effectively dead code.

Reply:
Yes, it is not needed. Could be called dead code. True for both functions that were recently added to this file.

Reply:
OK, I thought there might be a separate PR to fix this because this issue now exists in master.